### PR TITLE
Add OIDC role mapping

### PR DESCRIPTION
**Pull request by akhter-ali-idexx**

### CODE REVIEW

File: pkg/platform/resource_oidc_identity_mapping.go [hunk 1]
```diff
@@ -97,11 +97,11 @@ func (r *odicIdentityMappingResource) Schema(ctx context.Context, req resource.S
 						Required: true,
 						Validators: []validator.String{
 							stringvalidator.RegexMatches(
-								regexp.MustCompile(`^(applied-permissions\/admin|applied-permissions\/user|applied-permissions\/groups:.+)$`),
-								"must start with either 'applied-permissions/admin', 'applied-permissions/user', or 'applied-permissions/groups:'",
+								regexp.MustCompile(`^(applied-permissions\/admin|applied-permissions\/user|applied-permissions\/groups|applied-permissions\/roles:.+)$`),
```
Comment (alexhung) [inline pkg/platform/resource_oidc_identity_mapping.go:100]: The trailing `:` is missing for the `groups`.

**Review by alexhung — CHANGES_REQUESTED** [review]
Please also add test for the `roles` scope. See https://github.com/jfrog/terraform-provider-platform/blob/main/pkg/platform/resource_oidc_identity_mapping_test.go#L130 for example.

**Comment by akhter-ali-idexx** [comment]
> Please also add test for the `roles` scope. See https://github.com/jfrog/terraform-provider-platform/blob/main/pkg/platform/resource_oidc_identity_mapping_test.go#L130 for example.

Done

**Comment by alexhung** [comment]
> > Please also add test for the `roles` scope. See https://github.com/jfrog/terraform-provider-platform/blob/main/pkg/platform/resource_oidc_identity_mapping_test.go#L130 for example.
> 
> Done

@akhter-ali-idexx Excellent!

Can you also try to sign the CLA? The bot may fails but that's ok as long as you sign.

**Comment by akhter-ali-idexx** [comment]
> @akhter-ali-idexx See [test failures](https://github.com/jfrog/terraform-provider-platform/actions/runs/11259232965/job/31307944840) Screenshot 2024-10-09 at 9 52 08 AM

Wow, I completely missed that, lol.  Test execution added

**Comment by akhter-ali-idexx** [comment]
@alexhung Please do let me know if I've missed anything else, I'd like to use asap.  Thank you!

File: pkg/platform/resource_oidc_identity_mapping.go (hunk 1 above)
Comment (alexhung) [reply pkg/platform/resource_oidc_identity_mapping.go:100]: The `:` is still missing. Should be `applied-permissions\/groups:`

**Review by alexhung — APPROVED** [review]
Look good enough for now. I'll take this from here and get it to be in a releasable state.

**Pull request merged**